packages/vite/src/index.ts

```diff
@@ -126,18 +126,22 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
         const { transforms } = generateTransforms(config, transformFactories, srcURL.searchParams, logger)
         const { image, metadata } = await applyTransforms(transforms, img.clone(), pluginOptions.removeMetadata)
 
-        if (viteConfig.command === 'serve') {
-          const id = await generateImageID(srcURL, config, img)
-          generatedImages.set(id, image)
-          metadata.src = basePath + id
+        if (directives.has('inline')) {
+          metadata.src = `data:image/${metadata.format};base64,${(await image.toBuffer()).toString('base64')}`
         } else {
-          const fileHandle = this.emitFile({
-            name: basename(pathname, extname(pathname)) + `.${metadata.format}`,
-            source: await image.toBuffer(),
-            type: 'asset'
-          })
-
-          metadata.src = `__VITE_ASSET__${fileHandle}__`
+          if (viteConfig.command === 'serve') {
```

Comment:
you could probably combine this line with the previous line to make it an `else if`, which would reduce the number of lines and amount of indentation